### ISSUE TITLE
Issue tracker is invalid. You must specify your Pivotal Tracker API token and Project ID

### ISSUE THREAD

**Issue by DenisBY** [errbit/errbit#899]
I'm trying to add a new user to the App, but getting this when clicking on Update button:
```
    Your app could not be updated.

Dang. The following errors are keeping this from being a success.

    Issue tracker is invalid
    You must specify your Pivotal Tracker API token and Project ID
```

But I have `API token and Project ID` in the App already:
```
Started PUT "/apps/543692c77d180f5cfd000004" for 127.0.0.1 at 2015-06-16 10:03:04 +0000
Processing by AppsController#update as HTML
  Parameters: {"utf8"=>"✓", "authenticity_token"=>"jW****/8=", "app"=>{"name"=>"LViS", "repository_branch"=>"", "github_repo"=>"", "bitbucket_repo"=>"", "asset_host"=>"", "current_app_version"=>"", "notify_on_errs"=>"1", "email_at_notices"=>"1, 10, 100", "notify_on_deploys"=>"1", "notify_all_users"=>"0", "watchers_attributes"=>{"0"=>{"watcher_type"=>"user", "user_id"=>"543695427d180f5cfd000009", "id"=>"543692c77d180f5cfd000005"}, "1"=>{"watcher_type"=>"user", "user_id"=>"5436894a7d180f5cfd000001", "id"=>"543695897d180f5cfd00000e"}, "2"=>{"watcher_type"=>"user", "user_id"=>"5436a4127d180f03c3000001", "id"=>"5436a8467d180f03c300000a"}, "3"=>{"watcher_type"=>"user", "user_id"=>"544a47cf7d180f87b8000462", "id"=>"544a48067d180f87b8000466"}, "1434448979505"=>{"watcher_type"=>"user", "user_id"=>"557fee927d180f411e00000a"}}, "resolve_errs_on_deploy"=>"1", "issue_tracker_attributes"=>{"type_tracker"=>"pivotal", "options"=>{"api_token"=>"29*****19", "project_id"=>"5***3"}, "id"=>"557ff44a7d180f411e0000a3"}, "notification_service_attributes"=>{"type"=>"NotificationService", "id"=>"557ff44a7d180f411e0000a4"}}, "id"=>"543692c77d180f5cfd000004"}
```

Errbit shows `0.3.0.dev` at the bottom but:
```
$ git status
On branch 0-3-stable
Your branch is up-to-date with 'origin/0-3-stable'.
```

**Comment by stevecrozz** [comment]
Are you using the pivotal issue tracker integration? The error suggests your app is set to use this integration, but it isn't configured correctly. If that doesn't help, can you show me the output of:
bundle exec rails runner "puts App.find('543692c77d180f5cfd000004').inspect"

**Comment by DenisBY** [comment]
Yes, we use pivotal:
![2729](https://cloud.githubusercontent.com/assets/1147703/8201070/ac318954-14d5-11e5-963e-286b51391c2f.png)
```
$ bundle exec rails runner "puts App.find('543692c77d180f5cfd000004').inspect"
#<App _id: 543692c77d180f5cfd000004, created_at: 2014-10-09 13:51:03 UTC, updated_at: 2014-11-03 14:37:30 UTC, name: "My App", api_key: "eb**********6d", github_repo: "", bitbucket_repo: "", asset_host: "", repository_branch: "", current_app_version: "", resolve_errs_on_deploy: true, notify_all_users: false, notify_on_errs: true, notify_on_deploys: true, email_at_notices: [1, 10, 100]>
```

I also see "Your app could not be updated." right after I'm logging in to the errbit, w/o editing any App.

**Comment by DenisBY** [comment]
mongo:
```
> db.apps.find({"_id" : "543692c77d180f5cfd000004"})
{ "_id" : "543692c77d180f5cfd000004", "api_key" : "eb****6d", "asset_host" : "", "bitbucket_repo" : "", "created_at" : ISODate("2014-10-09T13:51:03.434Z"), "current_app_version" : "", "email_at_notices" : [  1,  10,  100 ], "github_repo" : "", "issue_tracker" : { "type_tracker" : "pivotal", "options" : { "api_token" : "29*****19", "project_id" : "5****3" }, "updated_at" : ISODate("2014-11-29T14:33:04.189Z"), "created_at" : ISODate("2014-11-29T14:33:04.189Z") }, "name" : "LViS", "notification_service" : { "_id" : ObjectId("558119a37d180f411e0000c5"), "notify_at_notices" : [  0 ], "_type" : "NotificationService" }, "notify_all_users" : false, "notify_on_deploys" : true, "notify_on_errs" : true, "repository_branch" : "", "resolve_errs_on_deploy" : true, "updated_at" : ISODate("2014-11-03T14:37:30.233Z"), "watchers" : [     {       "_id" : ObjectId("543692c77d180f5cfd000005"),   "email" : null,      "user_id" : ObjectId("543695427d180f5cfd000009") },     {       "_id" : ObjectId("543695897d180f5cfd00000e"),   "user_id" : ObjectId("5436894a7d180f5cfd000001"),       "email" : null },       {       "_id" : ObjectId("5436a8467d180f03c300000a"),        "user_id" : ObjectId("5436a4127d180f03c3000001"),       "email" : null },       {       "_id" : ObjectId("544a48067d180f87b8000466"),   "user_id" : ObjectId("544a47cf7d180f87b8000462"),       "email" : null } ] }
```

**Comment by stevecrozz** [comment]
Thanks @DenisBY, that extra info from mongo was a big help in tracking this down. I just added a commit to 0-3-stable to fix this issue.

By the way, I am looking for a few brave souls who are willing to adopt and maintain all the various issue tracker plugins. If you're interested in maintaining the pivotal plugin, let me know so we can talk about it.

**Comment by DenisBY** [comment]
Thank you!
No sure about 'maintain'. I'm a DevOps so I can report bugs in case I found any.

**Comment by stevecrozz** [comment]
No prob. I do appreciate the reports. I only mention it because official support for these trackers is no longer available in errbit v0.4.0 and beyond. I figured that's why you're still on v0.3.0. Nobody else has stepped in to maintain this particular plugin and make it work with newer errbits so if you need this plugin, you'll be stuck on v0.3.0 indefinitely.

**Comment by DenisBY** [comment]
Oh, I see. I'll ask our programmers if someone want to maintain this pluging. Thanks!
